### ISSUE TITLE
Will it also run on Raspberry Pi 4 ?

### ISSUE THREAD

**Issue by wdklotz** [ehough/docker-kodi#37]
>>>>This image should work on any Linux distribution with a functional Docker installation.

Will it also run on Raspberry Pi 4 ?

**Comment by mviereck** [comment]
You'll likely have to build the image yourself on the RasPi, so it will become an arm base instead of an amd64 base.

**Comment by gsantner** [comment]
you really should ask yourself if you need to run such a power hungry application that requires all acceleration it can get ... through containerization on a quite low end machine (compared to a PC).

**Comment by wdklotz** [comment]
Well you are right! I solved my problem by instralling KODI as X11 client on my desktop, i.e. KODI and and Desktop use the same Xserver. I found the neccessary information from [here](https://www.raspberrypi.org/forums/viewtopic.php?t=251645).

**Comment by rimago** [comment]
Hey, I have created a fork and modified it to run on Raspberry Pi 4 without x11docker. It is not fully tested, but you can have a look at:
[rimago/rpi-kodi](https://hub.docker.com/r/rimago/rpi-kodi)

Unfortunately it still requires the --privileged flag. I am open for suggestions on how to remove that.

**Comment by ngpsherwood** [comment]
Hello rimago I test it on my raspberry pi with docker and not working

It's a log if you want fix it 

https://pastebin.pl/view/06716247

**Comment by rimago** [comment]
Hey,
thank you for providing the logfile. It seems to be a permission problem with the home directory?
Did you add the kodi user to your host and make him as the owner of the home directory you mount in the container?

**Comment by wdklotz** [comment]
A humble question: How do you 'docker run ....' the image?
WDK




Am Di., 2. Feb. 2021 um 17:10 Uhr schrieb rimago <notifications@github.com>:

**Comment by rimago** [comment]
You can find an example docker-compose.yml in the Readme of the Git-Repository:
https://github.com/rimago/rpi-kodi

**Comment by ngpsherwood** [comment]
Hello I did and it's some thing  keep errors like post log

**Comment by wdklotz** [comment]
Hi. I started the container with 'docker-compose up' like told with docker-compose.yml. I see the gui kodi screen. But the container (host is rasberry pi 4) does not react on my mouse, therefore there is no way to control the kodi container.
Any hint available?

**Comment by rimago** [comment]
Good point. I have not attached a keyboard or mouse to my raspberry. I am controlling kodi with the web-interface and the Android app (Kore).
I guess if you want to use your mouse and keyboard you have to map these devices somehow into the container (extend the devices in the docker-compose.yml).

**Comment by rimago** [comment]
Btw. this issue is now splitting up into two different topics.
It might be useful if you could create separate issues (permission problem and no mouse/keyboard) in the fork:
https://github.com/rimago/rpi-kodi

@ngpsherwood : could you provide me your docker-compose.yml and the output of the following command:
ls -al /home/pi/kodi/home
(if this is the home directory you want to use)

**Comment by ngpsherwood** [comment]
Log say drwxr-xr-x 2 pi kodi 4069
drwxr-xr-x 3 pi pi 4069

**Comment by rimago** [comment]
Ok, it seems the user pi is still the owner. Try to change this with the following command:
sudo chown -R kodi:kodi /home/pi/kodi/home

**Comment by ngpsherwood** [comment]
I done it with 755 and what is password for ssh kodi because I want enable to kore

**Comment by rimago** [comment]
You don't need ssh. You can start a bash in the container with the following command:
docker exec -it kodi /bin/bash

However, to enable the webserver for Kore you do not have to do this in the container. As you have the home-directory on your host system you could create there the advancedsettings.xml

For example. I have on my host-system:
```
pi@raspberrypi:~/kodi/home/.kodi/userdata $ cat advancedsettings.xml 
<advancedsettings>
    <services>
        <esallinterfaces>true</esallinterfaces>
        <webserver>true</webserver>
        <zeroconf>true</zeroconf>
    </services>
</advancedsettings>
```

**Comment by ngpsherwood** [comment]
Hey all working fine and kore app 
Thanks you and can you enable issue tab on you github for any issue 😊

**Comment by rimago** [comment]
I am happy to hear that :-)

I enabled the issues and also the discussion. Thank you for the hint ;-)
